message_ix_models/model/transport/CHN_IND.py

```diff
@@ -0,0 +1,250 @@
+"""Retrieve transport activity data for China (Statistical Yearbook of the National
+Bureau of Statistics) and for India (iTEM)."""
+
+import numpy as np
+import pandas as pd
+from iam_units import registry
+from item import historical  # type: ignore [import-not-found]
+
+from message_ix_models.util import package_data_path
+
+UNITS = {
+    "Population": (1.0e-6, None, "dimensionless"),
+    "Vehicle Stock": (1.0e4, "vehicle", "thousand vehicle"),
+    "Passenger-Kilometers": (100, "megapkm", "gigapkm"),
+    "Freight Ton-Kilometers": (100, "megatkm", "gigatkm"),
+}
+
+
+SDMX_MAP = {
+    "REF_AREA": "ISO_code",
+    "VARIABLE": "Variable",
+    "VEHICLE": "Vehicle type",
+    "MODE": "Mode/vehicle type",
+    "UNIT": "Units",
+    "TIME_PERIOD": "Year",
+    "VALUE": "Value",
+}
+#: Files containing data for China ("name_of_file", rows_to_skip_from_the_bottom)
+FILES = {
+    "Passenger activity": ("CHN_activity-passenger.csv", 4),
+    "Vehicle stock civil": ("CHN_stock-civil.csv", 5),
+    "Vehicle stock private": ("CHN_stock-private.csv", 1),
+    "Freight activity": ("CHN_activity-freight.csv", 5),
+}
+
+POP_FILE = "CHN_IND_population.csv"
+
+# Rail sub-categories to be removed from Chinese dataset
+RAIL_SUB_CAT = [
+    "Freight Ton-Kilometers of National Railways(100 million ton-km)",
+    "Freight Ton-Kilometers of Local Railways(100 million ton-km)",
+    "Freight Ton-Kilometers of Joint-venture Railways(100 million ton-km)",
+    "Passenger-Kilometers of National Railways(100 million passenger-km)",
+    "Passenger-Kilometers of Local Railways(100 million passenger-km)",
+    "Passenger-Kilometers of Joint-venture Railways(100 million passenger-km)",
+]
+
+# Mapping of variables to vehicle modes
+FILL_VALUES = {
+    "Passenger-Kilometers": "Total passenger transport",
+    "Freight Ton-Kilometers": "Total freight transport",
+}
+
+
+# TODO: migrate to convert_units.py, as an extra argument: e.g. axis=1 being the
+#  current way (converting per var names in columns) or axis=0 (converting per rows
+#  as coded in _covert below).
+def _convert(group_df):
+    """ ""Convert units of a pandas.groupby object based on UNITS."""
+    info = UNITS.get(group_df["Variable"].unique()[0], None)
+    if not info:
+        return group_df
+    factor, unit_in, unit_out = info
+    qty = registry.Quantity(factor * group_df["Value"].values, unit_in).to(
+        unit_out or unit_in
+    )
+    return group_df.assign(Value=qty.magnitude, Units=qty.units)
+
+
+def split_variable(s):
```

Comment:
Would be nice to have type hints on the functions in this file, I think :)

Reply:
Some of these codes are (very) old and not actively used; if/when they are updated or replaced, the new code will be type hinted.